### ISSUE TITLE
Rework implementation for multiple admissions per cycle

### ISSUE THREAD

**Issue by alculquicondor** [kubernetes-sigs/kueue#1867]
**What would you like to be cleaned**:

Re-design the logic that allows multiple admissions to happen per cycle without incurring in over admission.

I think a simpler approach would be along the lines of modifying the snapshot (add admitted workloads and remove preempted workloads within the cycle) and checking if the incoming workload still fits. There is a special case for workloads needing preemption where there are no candidates for preemption. I think we can add a smaller fake workload that reserves space for the workload without going beyond the nominal quotas.

**Why is this needed**:

The current logic has been reworked multiple times with various pros and cons.

- #1866
- #1399
- #1024
- #805
- #475

Now it's very hard to follow.

**Comment by alculquicondor** [comment]
cc @yaroslava-serdiuk

**Comment by gabesaba** [comment]
in #2641, we enhance and simplify multiple preemption logic, by using the snapshot to account for usage. After promotion to beta in 0.9, we can delete the old code in 0.10

/close
